### PR TITLE
Add SSL support for the control app

### PR DESCRIPTION
**Pull request by composerinteralia**
This PR includes two commits, one for the refactor and one for the change. I can open two separate PRs if it is easier to review that way.

---

### Extract class for building SSL context
 
This commit extracts the `MiniSSL::Context` creation into its own
`MiniSSL::ContextBuilder` class along the same lines as in #1989.

This will allow us to reuse this code for adding SSL support to the
control app (issue #2015). Since we will need the `MiniSSL` require
and check in both places, I moved that into the `ContextBuilder` class
as well.

----

### Add SSL support for the control app

This starts to address #2015. I think we will need to add SSL support
to the control cli as well.

### CODE REVIEW

**Comment by composerinteralia** [comment]
Oops, I forgot to add a changelog entry. I will be sure to add that if this change looks OK.

File: lib/puma/runner.rb [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 require 'puma/server'
 require 'puma/const'
+require 'puma/minissl/context_builder'
```
Comment (nateberkopec) [inline lib/puma/runner.rb]: Nice call. I'm not gonna review that commit because it looks like a straight copy-paste into the new class.

**Comment by nateberkopec** [comment]
Nice! LGTM.

**Comment by composerinteralia** [comment]
Thanks! Changelog entry added.

**Pull request merged**